lib/plug/cowboy/handler.ex

```diff
@@ -15,12 +18,21 @@ defmodule Plug.Cowboy.Handler do
       {:ok, req, {plug, opts}}
     catch
       kind, reason ->
-        exit_on_error(kind, reason, System.stacktrace(), {plug, :call, [conn, opts]})
+        stacktrace = System.stacktrace()
+        metadata = %{kind: kind, error: reason, stacktrace: stacktrace}
+        measurements = %{duration: System.monotonic_time() - start}
+        :telemetry.execute([:plug_cowboy, :handler, :failure], measurements, metadata)
+
+        exit_on_error(kind, reason, stacktrace, {plug, :call, [conn, opts]})
     after
       receive do
         @already_sent -> :ok
       after
-        0 -> :ok
+        0 ->
+          measurements = %{duration: System.monotonic_time() - start}
+          :telemetry.execute([:plug_cowboy, :handler, :stop], measurements)
```

Comment:
Won't this send the `:stop` event in both the successful case & the exception case since it's in `after`? In my [PR](https://github.com/elixir-plug/plug_cowboy/pull/27) I put this call elsewhere so it wouldn't duplicate & so that we could send the `conn` in the metadata

Reply:
Yes, `:exception` should only be used for early exit/termination, for cases where program flow didn’t reach the point where a normal `:stop` event would be sent. `:exception` should always come with a stacktrace and reraise the exception back to the caller.